test/gtest/avgpool.hpp

```diff
@@ -0,0 +1,449 @@
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+#include "cpu_avgpool.hpp"
+#include "get_handle.hpp"
+#include "tensor_holder.hpp"
+#include "verify.hpp"
+#include <gtest/gtest.h>
+#include <iostream>
+#include <miopen/avgpool.hpp>
+#include <miopen/miopen.h>
+
+template <class T>
+inline std::ostream& operator<<(std::ostream& os, const std::vector<T>& v)
+{
+    os << '{';
+    for(int i = 0; i < v.size(); ++i)
+    {
+        if(i != 0)
+            os << ',';
+        os << v[i];
+    }
+    os << '}';
+    return os;
+}
+
+struct AvgPoolTestCase
+{
+    std::vector<size_t> input_dims;
+    std::vector<int64_t> kernel_size;
+    std::vector<int64_t> stride;
+    std::vector<int64_t> padding;
+    bool ceil_mode;
+    bool count_include_pad;
+    int64_t divisor_override;
+    bool is_contiguous = true;
+
+    friend std::ostream& operator<<(std::ostream& os, const AvgPoolTestCase& tc)
+    {
+        return os << " input_dims:" << tc.input_dims << " kernel_size:" << tc.kernel_size
+                  << " stride:" << tc.stride << " padding:" << tc.padding
+                  << " ceil_mode:" << tc.ceil_mode << " count_include_pad:" << tc.count_include_pad
+                  << " divisor_override:" << tc.divisor_override
+                  << " is_contiguous:" << tc.is_contiguous;
+    }
+
+    std::vector<size_t> GetInput() const { return input_dims; }
+    std::vector<size_t> ComputeStrides(std::vector<size_t> inputDim) const
+    {
+        if(!is_contiguous)
+            std::swap(inputDim.front(), inputDim.back());
+        std::vector<size_t> strides(inputDim.size());
+        strides.back() = 1;
+        for(int i = inputDim.size() - 2; i >= 0; --i)
+            strides[i] = strides[i + 1] * inputDim[i + 1];
+        if(!is_contiguous)
+            std::swap(strides.front(), strides.back());
+        return strides;
+    }
+};
+
```

Comment:
@long10024070 maybe we should put this function into an util file in gtest instead of duplicating it in every gtest files